crates/eww/src/config/system_stats.rs

```diff
@@ -7,32 +7,49 @@ use sysinfo::{ComponentExt, DiskExt, NetworkExt, NetworksExt, ProcessorExt, Syst
 
 static SYSTEM: Lazy<Mutex<System>> = Lazy::new(|| Mutex::new(System::new()));
 
-pub fn disk() -> String {
+pub fn get_disks() -> String {
     let mut c = SYSTEM.lock().unwrap();
     c.refresh_disks_list();
 
     format!(
         "{{ {} }}",
         c.get_disks()
             .iter()
-            .map(|c| format!(
-                r#""{}": {{"name": {:?}, "total": {}, "free": {}}}"#,
-                c.get_mount_point().display(),
-                c.get_name(),
-                c.get_total_space(),
-                c.get_available_space(),
-            ))
+            .map(|c| {
+                let total_space = c.get_total_space();
+                let available_space = c.get_available_space();
+                format!(
+                    r#""{}": {{"name": {:?}, "total": {}, "free": {}, "used": {}}}"#,
+                    c.get_mount_point().display(),
+                    c.get_name(),
+                    total_space,
+                    available_space,
+                    total_space - available_space,
+                )
+            })
             .join(",")
     )
 }
 
-pub fn ram() -> f32 {
+pub fn get_ram() -> String {
     let mut c = SYSTEM.lock().unwrap();
     c.refresh_memory();
-    (c.get_used_memory() as f32 + c.get_used_swap() as f32) / 1_000_000f32
+
+    let total_memory = c.get_total_memory();
+    let used_memory = c.get_used_memory();
+    format!(
+        r#"{{"total_mem": {}, "free_mem": {}, "total_swap": {}, "free_swap": {}, "available_mem": {}, "used_mem": {}, "used_mem_perc": {}}}"#,
+        total_memory,
+        c.get_free_memory(),
+        c.get_total_swap(),
+        c.get_free_swap(),
+        c.get_available_memory(),
+        used_memory,
+        used_memory / total_memory,
```

Comment:
This isn't how one calculates percentages,..... this needs *100 somewhere, and then rounding _after_ that, not before